searchlib/src/vespa/searchlib/util/statebuf.cpp

```diff
@@ -168,35 +145,28 @@ StateBuf::appendHex(unsigned long val) noexcept
 
 
 StateBuf &
-StateBuf::operator<<(const struct timespec &ts) noexcept
+StateBuf::operator<<(std::chrono::nanoseconds ns) noexcept
 {
-    (*this << static_cast<unsigned long>(ts.tv_sec) << '.').
-        appendDecFraction(static_cast<unsigned long>(ts.tv_nsec), 9);
+    std::chrono::seconds sec = std::chrono::duration_cast<std::chrono::seconds>(ns);
+    std::chrono::nanoseconds remainder = (ns - sec);
+    (*this << static_cast<unsigned long>(sec.count()) << '.').
+        appendDecFraction(static_cast<unsigned long>(remainder.count()), 9);
     return *this;
 }
 
 
 StateBuf &
-StateBuf::appendTimestamp(const struct timespec &ts) noexcept
+StateBuf::appendTimestamp(std::chrono::nanoseconds ns) noexcept
 {
-    appendKey("ts") << ts;
+    appendKey("ts") << ns;
     return *this;
 }
 
 
 StateBuf &
 StateBuf::appendTimestamp() noexcept
 {
-    struct timespec ts;
-    /*
-     * clock_gettime() is supposed to be async signal safe.
-     * gettimeofday() is not documented to be async signal safe.
-     */
-    int gtres = clock_gettime(CLOCK_REALTIME, &ts);
-    if (gtres != 0) {
-        LOG_ABORT("should not be reached");
-    }
-    appendTimestamp(ts);
+    appendTimestamp(std::chrono::system_clock::now().time_since_epoch());
```

Comment:
Is this method async signal safe ?

Reply:
It uses clock_gettime(CLOCK_REALTIME) under the hood if available.